src/kafka-file/languageservice/services/hover.ts

```diff
@@ -0,0 +1,168 @@
+import { Hover, Position, Range, TextDocument } from "vscode";
+import { createTopicDocumentation, SelectedClusterProvider, TopicProvider } from "../kafkaFileLanguageService";
+import { consumerModel, Model, producerModel } from "../model";
+import { Block, BlockType, Chunk, ConsumerBlock, KafkaFileDocument, MustacheExpression, NodeKind, ProducerBlock, Property } from "../parser/kafkaFileParser";
+
+export class KafkaFileHover {
+
+    constructor(private selectedClusterProvider: SelectedClusterProvider, private topicProvider: TopicProvider) {
+
+    }
+
+    async doHover(document: TextDocument, kafkaFileDocument: KafkaFileDocument, position: Position): Promise<Hover | undefined> {
+        // Get the AST node before the position where complation was triggered
+        const node = kafkaFileDocument.findNodeAt(position);
+        if (!node) {
+            return;
+        }
+        switch (node.kind) {
+
+            case NodeKind.consumerBlock: {
+                const block = <Block>node;
+                const topic = block.getPropertyValue('topic');
+                return new Hover(
+                    `Consumer declaration${topic ? ` for the topic \`${topic}\`` : ''}.\n\nSee [here](https://github.com/jlandersen/vscode-kafka/blob/master/docs/Consuming.md#kafka-file) for more informations.`,

```

Comment:
how about using the internal doc instead of opening some remote page? That way it could work offline

Reply:
`for topic`

Reply:
```typescript
const doc = new MarkdownString (`Consumer declaration${topic ? ` for topic \`${topic}\`` : ''}.\n\nSee [here](command:vscode-kafka.open.docs.page?%5B%7B%22page%22%3A%22Consuming%22%2C%22section%22%3A%22kafka-file%22%7D%5D) for more informations.`);
doc.isTrusted = true;
return new Hover(
                    doc,
                    node.range());
```

Reply:
Wow thanks so much for your great suggestion. I wanted to manage link to the embedded documentation but I was not able to do that. `doc.isTrusted = true;` is the solution. Thanks again!

Reply:
fixed